rclrs/src/node/mod.rs

```diff
@@ -40,6 +40,28 @@ pub struct Node {
     pub(crate) subscriptions: Vec<Weak<dyn SubscriptionBase>>,
 }
 
+/// A builder for node instantiation.
+///
+/// default value in each fields are
+///   - context: user defined
+///   - name: user defined
+///   - namespace: "/"
+///
+/// # Example
+/// ```
+/// # use rclrs::{Context, Node, NodeBuilder, RclReturnCode};
+/// let context = Context::new([])?;
+/// let node = NodeBuilder::new("foo_node", &context).namespace("/bar").build()?;
+/// assert_eq!(node.name(), "foo_node");
+/// assert_eq!(node.namespace(), "/bar");
+/// # Ok::<(), RclReturnCode>(())
+/// ```
+pub struct NodeBuilder<'a> {
+    pub(self) context: &'a Context,
+    pub(self) name: &'a str,
+    pub(self) namespace: &'a str,
```

Comment:
How about this? Then it doesn't need a lifetime, which I think is nice.

```suggestion
pub struct NodeBuilder<'a> {
    context_handle: Arc<Mutex<rcl_context_t>>,
    name: String,
    namespace: String,
```

I'd also remove `pub(self)`, since leaving it out has the same meaning.

Note that I'm not suggesting to change the function signatures – i.e. `NodeBuilder::new()` would still take a `&Context`, and `NodeBuilder::namespace()` a `&str` etc.

Reply:
I misunderstood field visibility of public struct.
I also agree not to use reference.
Is it ok to think that performance does not particularly important for builder?